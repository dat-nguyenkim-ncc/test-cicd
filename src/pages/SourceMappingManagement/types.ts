import { FieldTypes } from '../../components/TextField'

export type FundingRoundMappingDTO = {
  sourceValue: string
  round2Id: number
  round1Id: number
  round1: string
  round2: string
  id: number
}

export type FundingRoundMappingResponse = { data: FundingRoundMappingDTO[]; total: number }

export type GetFundingRoundMappingsResponse = {
  getFundingRoundMappings: FundingRoundMappingResponse
}

export type PendingFundingRoundMappingUpdate = Record<
  keyof Omit<FundingRoundMappingDTO, 'id'>,
  string | number
>

export type FundingRoundMappingFieldDTO = {
  field: keyof FundingRoundMappingDTO
  key: keyof FundingRoundMappingDTO
  label: string
  canEdit: boolean
  validate?: (v: string | number) => boolean
  format?: (v: string | number) => string
  fieldType: FieldTypes
  disablePopover?: boolean
  formLabel?: string
}

export type FundingRoundMappingFilterDTO = {
  sourceValue?: string
  round2Id: number
  round1Id: number
  isSourceValueBlank: boolean
}
