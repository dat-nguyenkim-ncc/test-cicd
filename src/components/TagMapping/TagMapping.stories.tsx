import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import TagMapping, { TagMappingProps } from './TagMapping'
import { Section } from '../primitives'

export default {
  title: 'Taxonomy/TagMapping',
  component: TagMapping,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<TagMappingProps> = args => (
  <Section sx={{ m: 4, margin: 'auto' }}>
    <TagMapping {...args} />
  </Section>
)

export const Default = Template.bind({})
Default.args = {
  title: 'Fintech',
  mappings: {
    aux: {
      data: [
        {
          id: '1602162955848',
          value: 'retail_banking',
          label: 'Retail Banking',
          endpoint: true,
          parent: [
            { id: '1602163039764', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162670040', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162616471', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162954943', label: 'Retail Banking', value: 'retail_banking' },
          ],
        },
      ],
    },
    primary: {
      data: [
        {
          id: '1602162955848',
          value: 'retail_banking',
          label: 'Retail Banking',
          endpoint: true,
          parent: [
            { id: '1602163039764', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162670040', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162616471', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162954943', label: 'Retail Banking', value: 'retail_banking' },
          ],
        },
        {
          id: '1602162955848',
          value: 'retail_banking',
          label: 'Retail Banking',
          endpoint: true,
          parent: [
            { id: '1602163039764', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162670040', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162616471', label: 'Corporate Banking', value: 'corporate_banking' },
            { id: '1602162954943', label: 'Retail Banking', value: 'retail_banking' },
          ],
        },
      ],
    },
  },
} as TagMappingProps
