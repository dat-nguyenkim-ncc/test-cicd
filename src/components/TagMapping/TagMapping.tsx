import React from 'react'
import { Box, Flex, Grid } from 'theme-ui'
import { Button, ButtonText, Triangle } from '..'
import { CompanyTypeSector, TagData, TaxonomyType, ViewInterface } from '../../types'
import strings from '../../strings'
import { Paragraph } from '../primitives'
import { ButtonTextProps } from '../ButtonText/ButtonText'
import {
  HasHistoryField,
  HasPendingCQField,
  ViewHistoryProps,
  ViewPendingChangeRequest,
} from '../../pages/CompanyForm/CompanyForm'
import ReasonPopover from '../ReasonPopover'
import { popoverZIndex } from '../../utils/consts'
import { ColumnNames, findCQ, TableNames } from '../../pages/CompanyForm/helpers'
import CompanyContext from '../../pages/CompanyForm/provider/CompanyContext'
import {
  EnumCompanyTypeSector,
  EnumDimensionType,
  EnumExpandStatusId,
  ETaxonomyKeys,
  TagTypes,
} from '../../types/enums'
import { Palette } from '../../theme'

const GRID = '125px 46px repeat(7, 115px 30px)'
const EXTRA_WIDTH = '200px'

export type TMapping = { data: TagData[]; buttons?: ButtonTextProps[] }

export type TagMappingProps = ViewInterface<{
  title: string
  typeTech: CompanyTypeSector
  mappings: {
    primary?: TMapping
    aux?: TMapping
    group?: TMapping
  }
  extras?: {
    primary?: TMapping
    aux?: TMapping
    group?: TMapping
  }
  onClickRemove?(type: TaxonomyType, typeTech: CompanyTypeSector, tag: TagData): void
  renderTags?(props: TagMappingGridProps): React.ReactElement
  viewTagsPendingCQFn?: (tag: TagData, columnName: string) => (() => void) | undefined
  viewTagsHistoryFn?: (tag: TagData, columnName: string) => (() => void) | undefined
  numberOfCRs?: number
  setTagPendingRequestType?(type: TagTypes | undefined): void
}> &
  Partial<ViewHistoryProps> &
  Partial<ViewPendingChangeRequest>

export type TagMappingItemProps = ViewInterface<{
  tag: TagData
  onClickRemove?(tag: TagData): void
  columns: string
  isTagMaps?: boolean
  viewTagsPendingCQFn?: (tag: TagData, columnName: string) => (() => void) | undefined
  viewTagsHistoryFn?: (tag: TagData, columnName: string) => (() => void) | undefined
}> &
  Partial<ViewHistoryProps> &
  Partial<ViewPendingChangeRequest>

export const TagMappingItem = ({
  tag,
  sx,
  onClickRemove,
  columns,
  isTagMaps,
  showViewHistory,
  showPendingChangeRequest,
  viewTagsHistoryFn,
  viewTagsPendingCQFn,
}: TagMappingItemProps) => {
  //Context
  const { overviewPendingRequest } = React.useContext(CompanyContext)

  const isAppendCQ = +(tag.fctStatusId || 0) === +EnumExpandStatusId.CHANGE_REQUEST
  const oldValue = tag.fctStatusId
  const SOURCE_NA = 'NA'

  const identity: HasHistoryField = {
    columnName: ColumnNames.FCT_STATUS_ID,
    tableName: TableNames.COMPANIES_TAGS,
    rowId: tag.rowId,
    source: SOURCE_NA as string,
  }

  const { total: numPending } = findCQ<HasPendingCQField>(overviewPendingRequest, identity) || {
    total: 0,
  }

  // const hasPendingRequest = !isAppendCQ && numPending > 0
  const hasHistory =
    !isAppendCQ &&
    showViewHistory?.(identity.tableName, identity.columnName, identity.rowId, identity.source)
  if (isTagMaps) {
    return (
      <>
        <ReasonPopover
          zIndex={popoverZIndex}
          buttons={[]}
          oldValue={oldValue}
          newValue={''}
          setReason={() => {}}
          label={hasHistory ? ' ' : undefined}
          viewHistory={isAppendCQ ? undefined : viewTagsHistoryFn?.(tag, ColumnNames.FCT_STATUS_ID)}
          viewPendingChangeRequest={undefined}
          totalItemPendingCR={numPending}
          disablePopover={true}
        >
          <Grid
            gap={0}
            columns={columns}
            sx={{
              mb: 4,
              p: 3,
              alignItems: 'center',
              borderRadius: 10,
              border: '1px solid black',
              position: 'relative',
              ...sx,
            }}
          >
            {(tag.parent || []).map(
              (p, index) =>
                tag.dimension === p.dimension && (
                  <React.Fragment key={index}>
                    <Paragraph>{p.label}</Paragraph>
                    <Triangle />
                  </React.Fragment>
                )
            )}
            <Paragraph>{tag.label}</Paragraph>
            {onClickRemove && (
              <Button
                sx={{ position: 'absolute', right: 0 }}
                variant="invert"
                size="normal"
                icon="remove"
                onPress={() => onClickRemove(tag)}
              />
            )}
          </Grid>
        </ReasonPopover>
      </>
    )
  }
  return (
    <Grid
      gap={0}
      columns={columns}
      sx={{
        mb: 3,
        p: 3,
        alignItems: 'center',
        borderRadius: 10,
        border: '1px solid black',
        position: 'relative',
        ...sx,
      }}
    >
      {(tag.parent || []).map(
        (p, index) =>
          tag.dimension === p.dimension && (
            <React.Fragment key={index}>
              <Paragraph>{p.label}</Paragraph>
              <Triangle />
            </React.Fragment>
          )
      )}
      <Paragraph>{tag.label}</Paragraph>
      {onClickRemove && (
        <Button
          sx={{ position: 'absolute', right: 0 }}
          variant="invert"
          size="normal"
          icon="remove"
          onPress={() => onClickRemove(tag)}
        />
      )}
    </Grid>
  )
}

export type TagMappingGridProps = ViewInterface<{
  onClickRemove?(tag: TagData): void
  columns?: string
  title?: string
  type: TaxonomyType
  copy: any
  mapping: TagData[]
  buttons?: ButtonTextProps[]
  renderTag?: (props: TagMappingItemProps) => React.ReactElement
  isTagMaps?: boolean
  viewTagsPendingCQFn?: (tag: TagData, columnName: string) => (() => void) | undefined
  viewTagsHistoryFn?: (tag: TagData, columnName: string) => (() => void) | undefined
}> &
  Partial<ViewHistoryProps> &
  Partial<ViewPendingChangeRequest>

export const TagMappingGrid = ({
  onClickRemove,
  columns = GRID,
  sx,
  mapping,
  title,
  type,
  buttons,
  isTagMaps,
  showViewHistory,
  showPendingChangeRequest,
  viewTagsHistoryFn,
  viewTagsPendingCQFn,
  renderTag = (props: TagMappingItemProps) => <TagMappingItem {...props} />,
}: TagMappingGridProps) => {
  return (
    <Box sx={sx}>
      {type !== ETaxonomyKeys.GROUP && title && (
        <Box mb={12}>
          {!!buttons?.length && (
            <Flex sx={{ justifyContent: 'flex-end', mb: 8 }}>
              {buttons?.map((b, idx) => (
                <ButtonText key={idx} {...b} />
              ))}
            </Flex>
          )}
          <Box sx={{ p: 3, bg: 'darkGray', borderRadius: 10 }}>
            <Paragraph bold sx={{ color: 'white' }}>
              {title}
            </Paragraph>
          </Box>
        </Box>
      )}
      {mapping?.length > 0 &&
        mapping.map((tag, row) => {
          return (
            <React.Fragment key={row}>
              {renderTag({
                tag,
                columns,
                onClickRemove,
                isTagMaps,
                showViewHistory,
                showPendingChangeRequest,
                viewTagsHistoryFn,
                viewTagsPendingCQFn,
              })}
            </React.Fragment>
          )
        })}
    </Box>
  )
}

const TagMapping = ({
  sx,
  title,
  typeTech,
  mappings,
  extras = {},
  onClickRemove,
  showViewHistory,
  viewTagsHistoryFn,
  viewTagsPendingCQFn,
  showPendingChangeRequest,
  numberOfCRs = 0,
  setTagPendingRequestType,
  renderTags = (props: TagMappingGridProps) => <TagMappingGrid {...props} />,
}: TagMappingProps) => {
  const { companyMapping: copy } = strings
  const subs = ['sub1', 'sub2', 'sub3', 'sub4', 'sub5']

  const buildMappings = (title: string, type: TaxonomyType) => {
    const mapping = mappings[type]?.data || []
    const buttons = mappings[type]?.buttons || []
    const extra = extras[type]?.data || []
    const extraBtns = extras[type]?.buttons || []
    const isTagMaps = type === ETaxonomyKeys.GROUP
    const isAux = type === ETaxonomyKeys.AUXILIARY

    if (mapping.length < 1 && extra.length < 1) return null
    return (
      <Box mt={3}>
        {isAux && typeTech === EnumCompanyTypeSector.FIN ? (
          <>
            <Flex>
              <Box sx={{ minWidth: EXTRA_WIDTH, maxWidth: EXTRA_WIDTH }}>
                <TagMappingGrid
                  title={`${title} ${
                    copy.dimension1ByTypeTech[typeTech as keyof typeof copy.dimension1ByTypeTech]
                  }${type === ETaxonomyKeys.AUXILIARY ? 'S' : ''}`}
                  type={type}
                  copy={copy}
                  mapping={[]}
                />
              </Box>
              <Box sx={!isTagMaps ? { flex: '1 1 auto', ml: 12 } : {}}>
                <TagMappingGrid
                  title={`${title} ${copy.dimension.cluster}${
                    type === ETaxonomyKeys.AUXILIARY ? 'S' : ''
                  }`}
                  type={type}
                  copy={copy}
                  mapping={[]}
                />
              </Box>
            </Flex>
            {mapping.map((mp, i) => {
              const sector = [(!mp.parent.length ? [mp] : mp.parent)[0]]
              const cluster =
                mp.dimensionType === EnumDimensionType.SECTOR
                  ? []
                  : [
                      {
                        ...mp,
                        parent:
                          mp.parent?.filter(p => p.dimensionType !== EnumDimensionType.SECTOR) ||
                          [],
                      },
                    ]
              return (
                <Flex key={i}>
                  <Box sx={{ minWidth: EXTRA_WIDTH, maxWidth: EXTRA_WIDTH }}>
                    {renderTags({
                      onClickRemove:
                        typeTech !== EnumCompanyTypeSector.FIN
                          ? tag => onClickRemove && onClickRemove(type, typeTech, tag)
                          : undefined,
                      columns: '',
                      type,
                      copy,
                      mapping: sector,
                      buttons: extraBtns,
                    })}
                  </Box>
                  <Box sx={!isTagMaps ? { flex: '1 1 auto', ml: 12 } : {}}>
                    {renderTags({
                      onClickRemove: tag => onClickRemove && onClickRemove(type, typeTech, tag),
                      columns: GRID,
                      type,
                      copy,
                      mapping: cluster,
                      buttons,
                      isTagMaps,
                      showViewHistory,
                      showPendingChangeRequest,
                      viewTagsHistoryFn,
                      viewTagsPendingCQFn,
                    })}
                  </Box>
                </Flex>
              )
            })}
          </>
        ) : (
          <Flex mt={2}>
            {!isTagMaps && (
              <Box sx={{ minWidth: EXTRA_WIDTH, maxWidth: EXTRA_WIDTH }}>
                {renderTags({
                  onClickRemove:
                    typeTech !== EnumCompanyTypeSector.FIN
                      ? tag => onClickRemove && onClickRemove(type, typeTech, tag)
                      : undefined,
                  columns: '',
                  type,
                  title: `${title} ${
                    copy.dimension1ByTypeTech[typeTech as keyof typeof copy.dimension1ByTypeTech]
                  }${type === ETaxonomyKeys.AUXILIARY ? 'S' : ''}`,
                  copy,
                  mapping: extra,
                  buttons: extraBtns,
                })}
              </Box>
            )}
            <Box sx={!isTagMaps ? { flex: '1 1 auto', ml: 12 } : {}}>
              {renderTags({
                onClickRemove: tag => onClickRemove && onClickRemove(type, typeTech, tag),
                columns: GRID,
                type,
                title: `${title} ${copy.dimension.cluster}${
                  type === ETaxonomyKeys.AUXILIARY ? 'S' : ''
                }`,
                copy,
                mapping,
                buttons,
                isTagMaps,
                showViewHistory,
                showPendingChangeRequest,
                viewTagsHistoryFn,
                viewTagsPendingCQFn,
              })}
            </Box>
          </Flex>
        )}
      </Box>
    )
  }

  const {
    pages: {
      addCompanyForm: { taxonomy },
    },
  } = strings

  return (
    <Box sx={sx}>
      <Flex sx={{ gap: 2, alignItems: 'center', justifyContent: 'space-between', ml: 0, mb: 4 }}>
        <Paragraph bold>{title}</Paragraph>
        {!!numberOfCRs && title === taxonomy.tagMapping.tagMaps && (
          <ButtonText
            onPress={() => {
              setTagPendingRequestType && setTagPendingRequestType(TagTypes.TAG)
            }}
            label={strings.common.viewPendingChangeRequest + ` (${numberOfCRs})`}
            sx={{ borderBottom: 0, color: Palette.orange, whiteSpace: 'nowrap' }}
          />
        )}
      </Flex>
      {!mappings.group && (
        <Grid gap={0} columns={`${EXTRA_WIDTH} auto`}>
          <Paragraph sx={{ flex: 1 }} bold>
            {copy.dimension1ByTypeTech[typeTech as keyof typeof copy.dimension1ByTypeTech]}
          </Paragraph>
          <Grid gap={0} columns={GRID} sx={{ alignItems: 'center', px: 3 }}>
            {subs.map((s, index) => (
              <React.Fragment key={s}>
                <Paragraph sx={{ flex: 1 }} bold>
                  {copy.headers[s as keyof typeof copy.headers]}
                </Paragraph>
                {index < subs.length - 1 && <Triangle />}
              </React.Fragment>
            ))}
          </Grid>
        </Grid>
      )}
      {buildMappings(copy.primary, 'primary')}
      {buildMappings(copy.aux, 'aux')}
      {buildMappings('', 'group')}
    </Box>
  )
}

export default TagMapping
