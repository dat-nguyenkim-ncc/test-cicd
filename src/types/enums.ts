export enum Routes {
  SEARCH = '/search',
  SEARCH_QUERY = '/search/:query',
  ADD_COMPANY = '/add',
  ADD_COMPANY_OVERVIEW = '/add/overview',
  ADD_COMPANY_FINANCIALS = '/add/financials',
  ADD_COMPANY_FUNDRAISING = '/add/fundraising',
  ADD_COMPANY_BUSINESS = '/add/business',
  ADD_COMPANY_PEOPLE = '/add/people',
  ADD_COMPANY_TAXONOMY = '/add/taxonomy',
  ADD_COMPANY_TAXONOMY_EXTERNAL = '/add/:id/taxonomy',
  COMPANY = '/company/:id',
  COMPANY_EDIT_SOURCE = '/company/sources/:id',
  COMPANY_NEW = '/company/new/:id/:source',
  EDIT_COMPANY_TAXONOMY = '/edit/:id/taxonomy',
  EDIT_COMPANY_OVERVIEW = '/edit/:id/overview',
  EDIT_COMPANY_FINANCIALS = '/edit/:id/financials',
  EDIT_COMPANY_INVESTMENTS = '/edit/:id/investments',
  EDIT_COMPANY_FUNDRAISING = '/edit/:id/fundraising',
  EDIT_COMPANY_BUSINESS = '/edit/:id/business',
  EDIT_COMPANY_PEOPLE = '/edit/:id/people',
  EDIT_COMPANY_ACQUISITIONS = '/edit/:id/acquisitions',
  EDIT_COMPANY_ACQUIREES = '/edit/:id/acquirees',
  EDIT_COMPANY_IPOS = '/edit/:id/ipos',
  EDIT_COMPANY_NEWS = '/edit/:id/news',
  EDIT_COMPANY_USE_CASE = '/edit/:id/use-case',
  EDIT_COMPANY_TECHNOLOGY = '/edit/:id/technology',
  EDIT_COMPANY_PARTNERSHIPS = '/edit/:id/partnerships',

  EDIT_COMPANY_FINANCIALS_CR = '/edit/:id/financials/:cr',
  EDIT_COMPANY_FUNDRAISING_CR = '/edit/:id/fundraising/:cr',
  EDIT_COMPANY_ACQUISITIONS_CR = '/edit/:id/acquisitions/:cr',
  EDIT_COMPANY_IPOS_CR = '/edit/:id/ipos/:cr',
  EDIT_COMPANY_NEWS_CR = '/edit/:id/news/:cr',
  EDIT_COMPANY_PEOPLE_CR = '/edit/:id/people/:cr',
  EDIT_COMPANY_TAXONOMY_CR = '/edit/:id/taxonomy/:cr',
  INVESTOR_MANAGEMENT_CR = '/investor-management/:cr',
  EDIT_COMPANY_USE_CASE_CR = '/edit/:id/use-case/:cr',
  EDIT_COMPANY_TECHNOLOGY_CR = '/edit/:id/technology/:cr',
  EDIT_COMPANY_BUSINESS_CR = '/edit/:id/business/:cr',
  EDIT_COMPANY_PARTNERSHIPS_CR = '/edit/:id/partnerships/:cr',

  MERGE = '/merge',
  MERGE_COMPANY = '/merge/:query',
  TAXONOMY_MANAGEMENT = '/taxonomy-management',
  TAG_MANAGEMENT = '/tag-management',
  MAPPING_ZONE = '/mapping-zone',
  REPORT_MANAGEMENT = '/report-management',
  COMPANY_MANAGEMENT = '/company-management',
  BULK_EDIT = '/company-management/bulk-edit',
  HEALTH_CHECK = '/health-check',
  CHANGE_REQUEST_MANAGEMENT = '/change-request-management',
  CHANGE_REQUEST_MANAGEMENT_TAXONOMY = '/taxonomy-change-request',
  INVESTOR_MANAGEMENT = '/investor-management',
  FIND_FINTECHS = '/find-fintechs',
  SIMILAR_COMPANIES = '/similar-companies',
  SOURCE_MAPPING_MANAGEMENT = '/source-mapping-management',
  INCORRECT_MAPPING = '/incorrect-mapping',
  ANALYSIS = '/analysis',
}

const { REACT_APP_STAGE } = process.env
const prefix = REACT_APP_STAGE === 'prod' ? 'prod' : 'dev'

export const EnumUserGroups = {
  KT: `fct_dataplatform_${prefix}_KT`,
  EXPAND: `fct_dataplatform_${prefix}_expand`,
  EVS: `fct_dataplatform_${prefix}_EVS`,
  MISSION_PLUS: `fct_dataplatform_${prefix}_mission_plus`,
}

export enum EnumVariantKeys {
  primary = 'primary',
  secondary = 'secondary',
  outline = 'outline',
  outlineWhite = 'outlineWhite',
  black = 'black',
  muted = 'muted',
  error = 'error',
  invert = 'invert',
}

export enum EnumCompanyTypeSector {
  FIN = 'fintech',
  INS = 'insurtech',
  REG = 'regtech',
  OUT = 'out',
}

export enum EnumDimensionType {
  SECTOR = 'sector',
  CLUSTER = 'cluster',
  VALUE_CHAIN = 'value chain',
  RISH = 'risk',
}

export type DimensionType =
  | EnumDimensionType.SECTOR
  | EnumDimensionType.CLUSTER
  | EnumDimensionType.VALUE_CHAIN
  | EnumDimensionType.RISH

export enum EnumExpandStatus {
  FOLLOWING = 'Following',
  UNFOLLOWED = 'Unfollowed',
  DUPLICATED = 'Duplicated',
  TO_BE_EVALUATED = 'ToBeEvaluated',
  CHANGE_REQUEST = 'ChangeRequest',
}

export enum EnumExpandStatusId {
  FOLLOWING = '1',
  UNFOLLOWED = '2',
  DUPLICATED = '3',
  TO_BE_EVALUATED = '4',
  CHANGE_REQUEST = '5',
}

export enum EnumCompanyNewsStatusId {
  FOLLOWING = '1',
  UNFOLLOWED = '0',
}

export enum EnumCompanyNewsStatus {
  FOLLOWING = 'Following',
  UNFOLLOWED = 'Unfollowed',
}

export enum EnumTagGroupSource {
  BCG = 'bcg',
  DR = 'dealroom',
  CRB = 'crunchbase',
  BCG_FIXED = 'bcg_fixed',
  FEEDLY = 'feedly',
  SWITCHPITCH = 'switchpitch',
}

export enum EnumApiSource {
  CRB = 'CRB',
  DR = 'DR',
  MANUAL = 'MANUAL',
  SWITCHPITCH = 'SWITCHPITCH',
  FEEDLY = 'FEEDLY',
}

export enum EnumReverseApiSource {
  CRB = 'crunchbase',
  DR = 'dealroom',
  MANUAL = 'bcg',
  SWITCHPITCH = 'switchpitch',
  FEEDLY = 'feedly',
}

export type ApiSourceType =
  | EnumApiSource.CRB
  | EnumApiSource.DR
  | EnumApiSource.MANUAL
  | EnumApiSource.SWITCHPITCH
  | EnumApiSource.FEEDLY

export enum RoundStatus {
  FOLLOWING = 'following',
  NOT_FOLLOWING = 'not following',
}

export enum EnumCompanySource {
  BCG = 'bcg',
  DR = 'dealroom',
  CRB = 'crunchbase',
}

export enum EnumReverseCompanySource {
  bcg = 'BCG',
  dealroom = 'DR',
  crunchbase = 'CRB',
}

export enum EnumInvestorSource {
  bcg = 'BCG',
  dealroom = 'DR',
  crunchbase = 'CRB',
  bcg_comp = 'BCG_COMP',
}

export enum EnumRemoveSourceType {
  MAKE_TO_OUT = 1,
  CREATE_NEW_MAPPING = 2,
  KEEP_EXISTING_MAPPING = 3,
  SET_AS_DUPLICATE = 4,
}

export enum EnumMergeStep {
  PRIMARY = 1,
  AUXILIARY = 2,
  FINTECH_TYPE = 3,
  TAGS = 4,
  PROFILE = 5,
  USE_CASE = 6,
  FUNDRAISING = 7,
  TECHNOLOGY = 8,
  LOCATION = 9,
  OVERRIDES = 10,
  SUMMARY = 11,
}

export enum ValidateInput {
  MAX_RANGE_BIGINT = 18446744073709551615,
  MAX_RANGE_INT = 2147483647,
}

export enum EnumDimensionCategories {
  FIN = 'fintech',
  INS = 'insurtech',
  REG = 'regtech',
}

export enum EnumDimensionValue {
  PRIMARY = 1,
  SECONDARY = 2,
}

export enum EnumS3Operation {
  PUT = 'putObject',
  GET = 'getObject',
}
export enum SortDirection {
  ASC = 'ASC',
  DESC = 'DESC',
}

export const PERMISSIONS: Partial<Record<Routes, string[]>> = {
  [Routes.TAG_MANAGEMENT]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS],
  [Routes.TAXONOMY_MANAGEMENT]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS],
  [Routes.REPORT_MANAGEMENT]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS],
  [Routes.COMPANY_MANAGEMENT]: [
    EnumUserGroups.EXPAND,
    EnumUserGroups.MISSION_PLUS,
    EnumUserGroups.EVS,
  ],
  [Routes.CHANGE_REQUEST_MANAGEMENT]: [
    EnumUserGroups.EXPAND,
    EnumUserGroups.MISSION_PLUS,
    EnumUserGroups.EVS,
  ],
  [Routes.INVESTOR_MANAGEMENT]: [
    EnumUserGroups.EXPAND,
    EnumUserGroups.MISSION_PLUS,
    EnumUserGroups.EVS,
  ],
  [Routes.MERGE_COMPANY]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS, EnumUserGroups.EVS],
  [Routes.COMPANY_EDIT_SOURCE]: [
    EnumUserGroups.EXPAND,
    EnumUserGroups.MISSION_PLUS,
    EnumUserGroups.EVS,
  ],
  [Routes.SOURCE_MAPPING_MANAGEMENT]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS],
  [Routes.ANALYSIS]: [EnumUserGroups.EXPAND, EnumUserGroups.MISSION_PLUS, EnumUserGroups.EVS],
}

export enum EnumAttachmentType {
  FCT_PROFILE = 'FCT profile',
  USE_CASE_PDF = 'Use case (PDF) - Live on portal',
  USE_CASE_PPT = 'Use case (PPT)',
  PRODUCT_PDF = 'Product (PDF) - Live on portal',
  FINANCIALS_PDF = 'Financials (PDF) - Live on portal',
  PITCH_DECK = 'Pitch deck',
  METHODOLOGY = 'Methodology',
  DEMO = 'Demo',
  OTHER = 'Other',
}

export enum TagTypes {
  FINTECHTYPE = 'FINTECHTYPE',
  TAG = 'TAG',
}

export enum EnumBoolean {
  TRUE = 1,
  FALSE = 0,
}

export enum EnumSourceNA {
  NA = 'NA',
}

export enum EnumInvestor {
  LEAD = 'Lead Investor',
  SUPPORT = 'Support Investor',
}

export enum ENumDataType {
  USE_CASE = 'use_cases',
  TECHNOLOGY = 'technologies',
  FUNDRAISING = 'fundraising',
}

export enum EnumSignUrlOperation {
  GET = 'getObject',
  PUT = 'putObject',
}

export enum EnumFileSize {
  PDF = 10485760,
  VIDEO = 52428800,
  IMG = 2097152,
}

export enum EPageKey {
  MAPPING_ZONE = 'mappingZone',
  FIND_FINTECHS = 'findFintechs',
  INCORRECT_MAPPING = 'incorrectMapping',
}

export enum ETaxonomyKeys {
  PRIMARY = 'primary',
  AUXILIARY = 'aux',
  GROUP = 'group',
}
