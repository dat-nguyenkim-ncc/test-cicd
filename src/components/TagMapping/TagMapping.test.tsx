import * as React from 'react'
import { shallow } from 'enzyme'
import TagMapping from './TagMapping'
import { EnumCompanyTypeSector } from '../../types/enums'

describe('TagMapping', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <TagMapping
        typeTech={EnumCompanyTypeSector.FIN}
        onClickRemove={() => {}}
        mappings={{
          primary: {
            data: [
              {
                id: '1602162955848',
                rowId: '123',
                value: 'retail_banking',
                label: 'Retail Banking',
                endpoint: true,
                description: 'mock',
                parent: [
                  {
                    id: '1602163039764',
                    label: 'Corporate Banking',
                    value: 'corporate_banking',
                    description: 'mock',
                  },
                  {
                    id: '1602162670040',
                    label: 'Corporate Banking',
                    value: 'corporate_banking',
                    description: 'mock',
                  },
                  {
                    id: '1602162616471',
                    label: 'Corporate Banking',
                    value: 'corporate_banking',
                    description: 'mock',
                  },
                  {
                    id: '1602162954943',
                    label: 'Retail Banking',
                    value: 'retail_banking',
                    description: 'mock',
                  },
                ],
              },
            ],
          },
        }}
        title="Fintech"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
