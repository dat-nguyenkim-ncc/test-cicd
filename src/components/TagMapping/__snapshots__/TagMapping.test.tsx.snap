// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagMapping matches snapshot 1`] = `ShallowWrapper {}`;
